sql/sql_base.cc

```diff
@@ -2245,6 +2245,43 @@ static TABLE_LIST* find_dup_table(THD *thd, const TABLE_LIST *table,
   DBUG_RETURN(res);
 }
 
+/**
+  Test that the subject table (of DELETE) has no cascade foreign key
+  parent present in the query.
+
+  @param  thd        thread handle
+  @param  table      table to be checked (must be updatable base table)
+  @param  table_list List of tables to check against
+
+  @retval non-NULL The table list element for the table that
+                   represents the cascade parent.
+  @retval NULL     No cascade parents found.
+*/
+TABLE_LIST *no_cascade_dependency(THD *thd, const TABLE_LIST *table,
```

Comment:
The actual TABLE_LIST* value is not used, the return value is immediately cast to bool, so return bool directly.
Consider removing "no_" from function name and return value logic